aptos-move/framework/aptos-framework/sources/Coin.move

```diff
@@ -228,25 +281,46 @@ module AptosFramework::Coin {
         amount: u64,
     ) acquires CoinInfo, CoinStore, MintCapability {
         let account_addr = Signer::address_of(account);
+
         assert!(
             exists<MintCapability<CoinType>>(account_addr),
             Errors::not_published(ENO_MINT_CAPABILITY),
         );
-        let _cap = borrow_global<MintCapability<CoinType>>(account_addr);
-        deposit(dst_addr, Coin<CoinType> { value: amount });
 
+        let cap = borrow_global<MintCapability<CoinType>>(account_addr);
+        let coins_minted = mint_with_capability<CoinType>(amount, cap);
+
+        deposit(dst_addr, coins_minted);
+    }
+
+    /// Mint new `Coin` with capability.
+    /// The capability `_cap` should be passed as reference to `MintCapability<CoinType>`.
+    /// Returns minted `Coin`.
+    public fun mint_with_capability<CoinType>(
+        amount: u64,
+        _cap: &MintCapability<CoinType>,
+    ): Coin<CoinType> acquires CoinInfo {
         let coin_addr = TypeInfo::account_address(&TypeInfo::type_of<CoinType>());
         let supply = &mut borrow_global_mut<CoinInfo<CoinType>>(coin_addr).supply;
         if (Option::is_some(supply)) {
             let supply = Option::borrow_mut(supply);
             *supply = *supply + amount;
-        }
+        };
+
+        Coin<CoinType> { value: amount }
     }
 
-    /// Registers to receive a specific CoinType. An account that wants to hold a coin type
```

Comment:
At this point in time, there's no clear value to separating these. Curious if you have thoughts on why we would want both?

Reply:
Sorry, maybe i misunderstood, but i through i had to remove MasteredCoin and move scripts back to coin?
If it's not so, for sure i will return back MasteredCoin

Reply:
If we have MasteredCoin we don't need two register for sure.
I just left them both because one is script, and another one is public function.

Reply:
> Let's keep the script functions in the original place, where possible

this one :)